vllm/forward_context.py

```diff
@@ -51,30 +51,25 @@ def get_forward_context() -> ForwardContext:
 @contextmanager
 def set_forward_context(attn_metadata: Any,
                         vllm_config: VllmConfig,
-                        virtual_engine: int = 0,
-                        num_tokens: int = 0):
+                        virtual_engine: int = 0):
     """A context manager that stores the current forward context,
     can be attention metadata, etc.
     Here we can inject common logic for every model forward pass.
     """
     global forward_start_time
-    need_to_track_batchsize = track_batchsize and attn_metadata is not None
```

Comment:
will `attn_metadata` be `None` in v0?

Reply:
Hmm.. I thought so when I checked the V0 flash-attn backend, but I'm not sure whether this is always the case.

Also, I just realized that this could also be a problem for the V1 TPU backend... Let me check.

Reply:
After second thought, I feel like maybe it makes sense to have `num_tokens` or other form of metadata as a forward context. Ideally, attention metadata should be just for attention.

Reply:
checked, neuron actually uses `None`:

https://github.com/vllm-project/vllm/blob/78ac0f591d8cde541ee74f2a0a8827d3769e2b86/vllm/worker/neuron_model_runner.py#L323

so maybe we should keep this line